### PR TITLE
chore: select 2.05 xenon testnet start height, dead code elimination

### PR DESCRIPTION
**Pull request by kantai**
This picks a 2.05 xenon/testnet start height, and does some dead code elimination

This implements #2889

### CODE REVIEW

File: src/core/mod.rs [hunk 1]
```diff
@@ -364,9 +343,15 @@ lazy_static! {
         StacksEpoch {
             epoch_id: StacksEpochId::Epoch20,
             start_height: BITCOIN_MAINNET_FIRST_BLOCK_HEIGHT,
-            end_height: STACKS_2_0_LAST_BLOCK_TO_PROCESS + 1,
+            end_height: BITCOIN_TESTNET_STACKS_2_05_BURN_HEIGHT,
```
Comment (kantai) [inline src/core/mod.rs]: Note that the prior value of end_height here would have made the `next-costs` branch broken when running on Xenon testnet: the end_height is long before the first bitcoin block height that the testnet considers

**Review by jcnelson — APPROVED** [review]
LGTM!  Do you also want to set the mainnet block height for 2.05 here in this PR, or do you want that to be separate?

**Comment by kantai** [comment]
> LGTM! Do you also want to set the mainnet block height for 2.05 here in this PR, or do you want that to be separate?

Done!

**Review by jcnelson — APPROVED** [review]

**Review by gregorycoppola — COMMENTED** [review]
@kantai can we get a clean change for the new height? And do the dead code elimination in separate PR? For easier review and possible future anthropology?

**Comment by gregorycoppola** [comment]
Oh i see Jude already lgtmed.. nevermind the idea of splitting in up if it's more overhead @kantai

**Review by gregorycoppola — APPROVED** [review]

File: src/core/mod.rs [hunk 2]
```diff
@@ -91,11 +88,13 @@ pub const BITCOIN_MAINNET_FIRST_BLOCK_TIMESTAMP: u32 = 1610643248;
 pub const BITCOIN_MAINNET_FIRST_BLOCK_HASH: &str =
     "0000000000000000000ab248c8e35c574514d052a83dbc12669e19bc43df486e";
 pub const BITCOIN_MAINNET_INITIAL_REWARD_START_BLOCK: u64 = 651389;
+pub const BITCOIN_MAINNET_STACKS_2_05_BURN_HEIGHT: u64 = 713_000;
 
 pub const BITCOIN_TESTNET_FIRST_BLOCK_HEIGHT: u64 = 2000000;
 pub const BITCOIN_TESTNET_FIRST_BLOCK_TIMESTAMP: u32 = 1622691840;
 pub const BITCOIN_TESTNET_FIRST_BLOCK_HASH: &str =
     "000000000000010dd0863ec3d7a0bae17c1957ae1de9cbcdae8e77aad33e3b8c";
+pub const BITCOIN_TESTNET_STACKS_2_05_BURN_HEIGHT: u64 = 2_104_076;
```
Comment (gregorycoppola) [inline src/core/mod.rs:97]: Is this in the past? @CharlieC3 https://www.blockchain.com/btc-testnet/block/0000000000000017bd07ae4a153830268e39d744efce605c69af5ccf6adf79a7

Comment (CharlieC3) [reply src/core/mod.rs:97]: Seems like it, is that intentional @kantai ?

Comment (kantai) [reply src/core/mod.rs]: Yep -- good catch -- updated to 2104380 -- that's the current block height (2104096) + 284, which should be about 48 hours.

**Pull request merged**